cmd/ingress-shim/controller/sync.go

```diff
@@ -102,13 +105,65 @@ func (c *Controller) buildCertificates(ing *extv1beta1.Ingress) ([]*v1alpha1.Cer
 				},
 			},
 		}
+
 		err = c.setIssuerSpecificConfig(crt, issuer, ing, tls)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
+		}
+
+		// check if a Certificate for this TLS entry already exists, and if it
+		// does then skip this entry
+		if existingCrt != nil {
+			glog.Infof("Certificate %q for ingress %q already exists", tls.SecretName, ing.Name)
+
+			if crtEqual(existingCrt, crt) {
+				glog.Infof("Certificate %q for ingress %q is up to date", tls.SecretName, ing.Name)
+				continue
+			}
+
+			updateCrt := existingCrt.DeepCopy()
+
+			updateCrt.Spec.DNSNames = tls.Hosts
+			updateCrt.Spec.SecretName = tls.SecretName
+			updateCrt.Spec.IssuerRef.Name = issuerName
+			updateCrt.Spec.IssuerRef.Kind = issuerKind
+			updateCrts = append(updateCrts, updateCrt)
+		} else {
+			newCrts = append(newCrts, crt)
 		}
-		crts = append(crts, crt)
 	}
-	return crts, nil
+	return newCrts, updateCrts, nil
+}
+
+// crtEqual checks and returns true if two Certificates are equal
+func crtEqual(a, b *v1alpha1.Certificate) bool {
```

Comment:
Can we call this `certNeedsUpdate` instead of `crtEqual` (i.e. flip the boolean)

Reply:
Done